input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+							State:                    util.StringCustomTypePtrToString(v.Properties.State),
+							SubscriptionID:           config.AzureSubscriptionID,
+							ResourceGroup:            resourceGroup,
+							ResourceType:             resourceType,
+							ResourceName:             resourceName,
+							CitusVersion:             util.StringPtrToString(v.Properties.CitusVersion),
+							HighAvailabilityEnabled:  util.BoolPtrToBool(v.Properties.EnableHa),
+							CoordinatorStorageMB:     util.Int32PtrToInt(v.Properties.CoordinatorStorageQuotaInMb),
+							NodeStorageMB:            util.Int32PtrToInt(v.Properties.NodeStorageQuotaInMb),
+							CoordinatorVCores:        util.Int32PtrToInt(v.Properties.CoordinatorVCores),
+							NodeVCores:               util.Int32PtrToInt(v.Properties.NodeVCores),
+							CoordinatorServerEdition: util.StringPtrToString(v.Properties.CoordinatorServerEdition),
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if resourceID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to find the database server info")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
+	client, err := azquery.NewMetricsClient(credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a metrics client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a metrics client: %v\n", err)
+		return
+	}
+
+	// Query metrics data with 1 min interval, for the last 1 min (should return 1 value)
+	metricNames := "cpu_percent,memory_percent,network_bytes_egress,network_bytes_ingress,storage_used"
+	if strings.ToLower(system.Info.Azure.ResourceType) == "flexibleserver" {
+		// metrics only available with Flexible Server
+		metricNames += ",txlogs_storage_used,read_iops,write_iops,disk_queue_depth,read_throughput,write_throughput"
+	}
+	option := &azquery.MetricsClientQueryResourceOptions{
+		MetricNames: to.Ptr(metricNames),
+		Aggregation: to.SliceOfPtrs(azquery.AggregationTypeAverage),
+		Interval:    to.Ptr("PT1M"),
+		Timespan:    to.Ptr(azquery.TimeInterval("PT1M")),
+	}
+
+	metricsRes, err := client.QueryResource(ctx, resourceID, option)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting server metrics: %v", err)
+		logger.PrintError("Azure/System: Encountered error getting server metrics: %v\n", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
```

Comment:
I actually saw that the only Reader role is enough to query this API too. Likely the Reader role is a parent role of the Monitoring Reader, so maybe this error message is incorrect and I should be simply saying "Reader permission" 🤔

Reply:
:man_shrugging:

Reply:
Just confirmed that "Monitoring Reader" is actually the required permission here, and "Reader" is too wide. I'll update and just mention "Monitoring Reader" here.